network/status.go

```diff
@@ -102,6 +102,10 @@ func (status *status) handleMessage(peer peer.ID, msg *StatusMessage) {
 		// update peer status message
 		status.peerMessage[peer] = msg
 
+		// check if message arrived is a block bigger then our latest one
+		// if so, fallback
```

Comment:
should be if >1 blocks farther

Reply:
sure

Reply:
Not sure how to interpret your comment here. 
Could you please give a hint on how to obtain our the actual block number inside this func ? 
I need to have something to compare against msg.BestBlockNumber, is this a correct assumption ?

Reply:
This should probably not be done in the status submodule. Anything to do with syncing should be handled at the root level of the network service and anything in the status submodule should only be concerned with the status message exchange.

Reply:
we should probably just add an interface function into status that'll allow us to request blocks if needed, not sure how that'll work but let's open an issue for it. I think it's out of scope for this

Reply:
We should not be adding anything to the status submodule. `status.handleMessage` should only be handling status messages, not block request messages.

Reply:
ok I raised #637 in order to try to address this comments, would be a good idea if we can jump into a call and get more details about the issue as I am not understanding exactly the changes requested.